### DISCUSSION TITLE
Distinguish scanned pdf or unscanned pdf

### DISCUSSION THREAD

**Discussion by Byshev333** [jsvine/pdfplumber#1100] [Q&A]
I would like to ask how to distinguish whether a pdf file is a scanned pdf or a non-scanned pdf? Some scanned pdfs are just a picture. After some scanned pdfs are recognized by ocr, a transparent text is covered on the original pdf. This makes it impossible to judge whether it is a scanned pdf by whether there is text output.

**Comment by jsvine** [comment]
Generally speaking, the `page.char` objects will have very different characteristics for scanned vs. non-scanned PDFs. Specifically, the scanned PDFs will have less-detailed `.chars` information. Another place you can look is the PDF's metadata (`pdf.metadata`).
